cmake/RepositoryDependenciesSetup.cmake

```diff
@@ -10,4 +10,78 @@ if (TPL_ENABLE_CUDA AND NOT Kokkos_ENABLE_Cuda_Relocatable_Device_Code)
   elseif (${PROJECT_NAME}_ENABLE_ShyLU_NodeTacho)
     message(FATAL_ERROR "ERROR: ${PROJECT_NAME}_ENABLE_ShyLU_NodeTacho=ON but TPL_ENABLE_CUDA='${TPL_ENABLE_CUDA}' AND Kokkos_ENABLE_Cuda_Relocatable_Device_Code='${Kokkos_ENABLE_Cuda_Relocatable_Device_Code}' which is not allowed!")
   endif()
-endif() 
\ No newline at end of file
+endif() 
+
+#########################################################################
+# STKBalance does not work with GO=INT or GO=UNSIGNED
+# Note and disable it if it was set as a dependence of STK
+# Error out if it was explicitly requested by user
+
+IF ((NOT ("${Tpetra_ENABLE_DEPRECATED_CODE}" STREQUAL "")) # Remove this test
```

Comment:
I think the usual idiom is this:
```
IF ((DEFINED Tpetra_ENABLE_DEPRECATED_CODE) AND (NOT Tpetra_ENABLE_DEPRECATED_CODE))
  # stuff ...
ENDIF()
```

Reply:
> I think the usual idiom is this:

@mhoemmen, that is actually not correct as explained in:

* https://tribits.org/doc/TribitsDevelopersGuide.html#how-to-check-for-and-tweak-tribits-enable-cache-variables

which says:

> First, note that most of these particular ENABLE variables are not BOOL cache variables but are actually STRING variables with the possible values of ON, OFF and empty "" (see the macro `SET_CACHE_ON_OFF_EMPTY()`). Therefore, just because the value of a `<XXX>_ENABLE_<YYY> `variable is defined (e.g. `IF (DEFINED <XXX>_ENABLE_<YYY>) ... ENDIF())` does not mean that it has been set to ON or OFF yet (or any non-empty values that evaluates to true or false in CMake). To see if an ENABLE variable is one of these variables, look in the CMakeCache.txt file for the type. If the type is STRING, then it is most likely this type of variable with a default value of empty "". However, if the cache type is `BOOL` then it is likely a standard bool variable that is not allowed to have a value of empty "".

Reply:
uh oh never mind then :)